### PR TITLE
Refactor for Chef Server 12

### PR DESCRIPTION
**Pull request by jtimberman**
This commit refactors the cookbook for Chef Server 12. The README
covers the new scope of this cookbook. We’ve attempted to preserve
parity in platform support with that of Chef Server 12 itself.

* Remove the omnitruck API client library and related attributes
* Use packagecloud repository for packages through the
chef_server_ingredient resource
* Remove the “dev” recipe
* Remove the dependency on the git cookbook
* Remove the Vagrantfile - we use test-kitchen now
* Update the Berksfile accordingly
* Add ServerSpec tests

### CODE REVIEW

**Comment by jtimberman** [comment]
Paging for review @opscode-cookbooks/community-engineering @opscode-cookbooks/ociv @opscode-cookbooks/server-team @stevendanna @paulmooring @mivok

**Comment by scotthain** [comment]
:+1:

**Comment by luckymike** [comment]
Just want to note that this update introduces a dependency tree, which makes bootstrapping with chef-solo potentially difficult. The current tree isn't that onerous, but if chef-server-ingredient or packagecloud gets new dependencies, it's going to become unmaintainable very quickly.

File: README.md [hunk 1]
```diff
@@ -159,93 +66,27 @@ system using this cookbook:
     # GO GO GO!!!
```
Comment (luckymike) [inline README.md]: You need to install the dependencies here, too.

Comment (jtimberman) [inline README.md]: Good point, will get the README updated.

**Comment by charlesjohnson** [comment]
Tested this out on the base kitchen.yml. :+1:

**Comment by jtimberman** [comment]
Note:

CentOS 7 is not yet supported. Internally we aren't building/testing on the platform yet, so we aren't publishing packages specifically for EL7. Users who want to try it out can use the following yum configuration for packagecloud:

```
[chef_stable_]
name=chef_stable_
baseurl=https://packagecloud.io/chef/stable/el/6/$basearch
repo_gpgcheck=1
gpgcheck=0
enabled=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-packagecloud_io
sslverify=1
sslcacert=/etc/pki/tls/certs/ca-bundle.crt
```

That is, this is the same packagecloud yum configuration as CentOS 6 - the package works and `chef-server-ctl test` succeeds. However, it is better to wait until we have packages built/tested specifically on EL7 so we don't fill the chef-server or chef-server-ingredient cookbooks with a bunch of workarounds we'll have to remove later.

**Comment by jtimberman** [comment]
@luckymike How's the README.md look to you in  3393cfb?

**Comment by luckymike** [comment]
Looks fine, but it still makes me nervous :). One thing though, you're chef-server download is from github, and the others are from the supermarket. Maybe standardize?

**Comment by jtimberman** [comment]
Fair point for consistency. Updated.

**Pull request merged**